### DISCUSSION TITLE
Error using Firebase-Admin in Edge Functions

### DISCUSSION THREAD

**Discussion by amed901107** [supabase/supabase#27885] [Questions]
import { initializeApp, cert, ServiceAccount, App, getApps } from "npm:firebase-admin@12.2.0/app"
import { getFirestore } from "npm:firebase-admin@12.2.0/firestore"
import { getDatabase } from "npm:firebase-admin@12.2.0/database"
import { getAuth } from 'npm:firebase-admin@12.2.0/auth';
Using these libraries in an edge function, the edge function is deployed correctly, but when I launch the request through postman it returns {
"code": "NOT_FOUND",
"message": "Requested function was not found"
} 404, using this variant
import { initializeApp, cert, ServiceAccount, App, getApps } from "https://esm.sh/firebase-admin@12.2.0/app"
import { getFirestore } from "https://esm.sh/firebase-admin@12.2.0/firestore"
The deploy throws me the following error
Error: failed to load 'https://esm.sh/error.js?type=unsupported-node-builtin-module&name=child_process&importer=google-auth-library@9.10.0': Import 'https://esm.sh/error.js?type=unsupported-node-builtin-module&name=child_process&importer=google-auth-library@9.10.0' failed: 500 Internal Server Error

**Comment by GaryAustin1** [comment]
As mentioned in your question in Discord, seems like the modules are not supported.
https://docs.deno.com/deploy/api/runtime-node/  Talks about the supported built in node modules.
And says child_process is NOT supported.
![image](https://github.com/supabase/supabase/assets/54564956/07138453-cce7-448b-a906-45d42e7165ad)

Edit: note this is just based on a search of your error and the hit in that Deno page.  Supabase does have their own implementation, but does depend on the base Deno repository for their code.

**Reply by amed901107** [reply]
Hi, ok but using the libraries in this format does not give error in the deploy
import { initializeApp, cert, ServiceAccount, App, getApps } from "npm:firebase-admin@12.2.0/app"
import { getFirestore } from "npm:firebase-admin@12.2.0/firestore"
import { getDatabase } from "npm:firebase-admin@12.2.0/database"
import { getAuth } from 'npm:firebase-admin@12.2.0/auth';
I run the local edge and it works, but online it responds 404
